File: algos/binary_search.py
def binary_search(input_list: list, target: int) -> bool:
	start: int = 0
	end: int = len(input_list)
	while (end > start):
		middle: int = int((end + start)/2)

		if target == input_list[middle]:
			return True
		elif target < input_list[middle]:
			end = middle
		else:
			start = middle + 1
	return False

input_list = [3, 2, 10, 11, 33, 27]
input_list.sort()
print()
print(binary_search(input_list, 33))

File: algos/binary_search.go
package main

import (
    "fmt"
)

func binary_search(array []int, target int) bool {
    start := 0
    end := len(array)
    for start < end {
        middle := start + (end-start)/2

        if array[middle] == target { return true }

        if array[middle] > target { 
            end = middle 
        } else { 
            start = middle + 1 
        }

    }
    return false
}

func main() {
    fmt.Println(binary_search([]int{1, 3, 11, 12, 14, 18, 18, 19}, 14))
}
